nixos/modules/tasks/filesystems.nix

```diff
@@ -41,11 +41,14 @@ let
       };
 
       options = mkOption {
-        default = "defaults";
-        example = "data=journal";
-        type = types.commas; # FIXME: should be a list
+        default = [ "defaults" ];
+        example = [ "data=journal" ];
         description = "Options used to mount the file system.";
-      };
+      } // (if versionAtLeast (readFile "${toString pkgs.path}/.version") "16.09" then {
```

Comment:
You can give `options` an `apply` attribute to map strings to lists so you don't have check the type elsewhere.